services/worker/src/worker/job_runners/_cached_based_job_runner.py

```diff
@@ -0,0 +1,63 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 The HuggingFace Authors.
+
+import json
+import random
+import re
+from hashlib import sha1
+from pathlib import Path
+from typing import Optional
+
+from libcommon.processing_graph import ProcessingStep
+from libcommon.storage import init_dir, remove_dir
+from libcommon.utils import JobInfo
+
+from worker.config import AppConfig
+from worker.job_runner import JobRunner
+
+
+class CachedBasedJobRunner(JobRunner):
+    """Base class for job runners that use a cached directory."""
+
+    base_cache_directory: Path
+    cache_subdirectory: Optional[Path] = None
+
+    def __init__(
+        self,
+        job_info: JobInfo,
+        app_config: AppConfig,
+        processing_step: ProcessingStep,
+        cache_directory: Path,
+    ) -> None:
+        super().__init__(
+            job_info=job_info,
+            app_config=app_config,
+            processing_step=processing_step,
+        )
+        self.base_cache_directory = cache_directory
+
+    def get_cache_subdirectory(self, digits: int = 14) -> str:
+        random_str = f"{random.randrange(10**(digits - 1), 10**digits)}"  # nosec B311
+        # TODO: Refactor, need a way to generate payload based only on provided params
```

Comment:
do we need this comment? what is the idea?

Reply:
I copied it from the previous implementation, currently, it generates the payload from job_info (dataset, config, and split) but I thought that in the future we could have a better way to generate it dynamically. 
But I can remove it does not makes sense.